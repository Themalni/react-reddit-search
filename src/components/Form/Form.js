import styled from 'styled-components'

const Form = styled.form`
  width: 90%;

  ${(props) => props.theme.media.tablet} {
    width: auto;
  }
`

export default Form