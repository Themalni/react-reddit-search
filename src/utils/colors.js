export const colors = {
  white: '#fff',
  graphite: '#666',
  grey: '#8a96a3',
  yellowOrange: '#ff9933',
  orange: '#ed5f55',
  orangeDark: '#d8574e',
  blue: '#3a81b7',
  blueLight: '#64a0ce'
};

// d02216