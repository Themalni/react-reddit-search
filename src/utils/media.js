export const media = {
  phone: '@media (min-width: 320px)',
  tablet: '@media (min-width: 728px)',
  laptop: '@media (min-width: 1024px)',
  desktop: '@media (min-width: 1280px)'
}