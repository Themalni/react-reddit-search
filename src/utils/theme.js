import { colors } from './colors';

export const theme = {
  colors,
  font: {
    thin: 300,
    regular: 400,
    bold: 700,
  },
  media: {
    phone: '@media (min-width: 320px)',
    tablet: '@media (min-width: 728px)',
    laptop: '@media (min-width: 1024px)',
    desktop: '@media (min-width: 1280px)'
  }
}