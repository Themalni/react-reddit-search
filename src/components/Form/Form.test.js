import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { theme } from '../../utils/theme';
import 'jest-styled-components'
import Form from './Form';
import { ThemeProvider } from 'styled-components';

describe('Form', () => {

  it("renders correctly", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Form></Form>
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot();
  })

})