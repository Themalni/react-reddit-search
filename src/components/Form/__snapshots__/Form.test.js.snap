// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly 1`] = `
.c0 {
  width: 90%;
}

@media (min-width:728px) {
  .c0 {
    width: auto;
  }
}

<form
  class="c0"
/>
`;
